java/debugger.jpda.truffle/src/org/netbeans/modules/debugger/jpda/truffle/frames/models/DebuggingTruffleActionsProvider.java

```diff
@@ -96,20 +122,60 @@ public void performDefaultAction(NodeActionsProvider original, Object node) thro
             }
             return actions;*/
         } else {
-            return original.getActions(node);
+            Action[] actions = original.getActions(node);
+            if (node instanceof DebuggingView.DVThread && hasGuestInfo(node)) {
+                Action showAction;
+                if (!isShowAllHostFrames((DebuggingView.DVThread) node)) {
+                    showAction = SHOW_HOST_ACTION;
```

Comment:
There is going to be an action "Show/Hide Java Frames", great!

Reply:
:-)